### PR TITLE
Fixes #678, #680, #684

### LINKED ISSUE #678: Answers to Questions are truncated at 255 characters
### What happened just before the problem occurred
I posted an answer to a question

### Relevant URLs

https://publiclab.org/questions/shubham/08-17-2016/how-to-neutralize-remove-the-source-s-spectra-to-obtain-absorption-spectra
### PublicLab.org username
cfastie
(to help reproduce the issue)

### Browser, version, and operating system
chrome, windows

### LINKED ISSUE #680: Submitting an answer to a question opened a note from 2010
### What happened just before the problem occurred
I clicked submit to post an answer to a question. I was taken to this note:
https://publiclab.org/notes/warren/12-14-2010/plant-spectral-timelapse-7hrs
 with this error:
"Only the author of the comment can edit it."

### Relevant URLs

https://publiclab.org/questions/shubham/08-17-2016/how-to-neutralize-remove-the-source-s-spectra-to-obtain-absorption-spectra

### PublicLab.org username
cfastie

### PR DESCRIPTION
**Pull request by ananyo2012**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] All tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [x] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

Fix error message display for comments

### CODE REVIEW

**Comment by jywarren** [comment]
Is this an urgent fix? I'm happy to pull it in.

**Comment by ananyo2012** [comment]
Yes it is a little urgent as the problem still persists. But please wait for my confirmation as I intend to add some other fixes in this PR too.

**Comment by ananyo2012** [comment]
@jywarren I think this can be merged now. It fixes three Issues!

**Comment by jywarren** [comment]
Oh OK I'll merge this one and publish -- can you close the related issues? I'll do this since one more person saw their answer cut off.

**Comment by jywarren** [comment]
Hm, i see one error -- 

```
Failure:
  <nil> expected but was
  <"The comment could not be saved.">.
test_should_show_error_if_answer_comment_not_saved(CommentControllerTest)
test/functional/comment_controller_test.rb:68:in `block in <class:CommentControllerTest>'
```

I'm happy to merge this if you can get that one to pass. Thanks!

File: db/migrate/20160818201147_change_content_column_type_in_answers.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class ChangeContentColumnTypeInAnswers < ActiveRecord::Migration
+  def up
+  	change_column :answers, :content, :text, limit: 2147483647
```
Comment (jywarren) [inline db/migrate/20160818201147_change_content_column_type_in_answers.rb]: Also, can you put in two spaces here (and elsewhere) instead of a tab? Thanks!

**Comment by ananyo2012** [comment]
@jywarren I accidentally changed some other lines. That's why the test was failing. This is solved now. Ready for merge!

**Comment by ananyo2012** [comment]
@jywarren 15e0d68 fixes a problem which is happening now. The case when no results are found in a tag sort. I have also added a test for this.

**Comment by jywarren** [comment]
Running tests...

**Comment by jywarren** [comment]
Passed -- rebasing and merged in e2371d4!

**Pull request closed without merging**